tests/phpunit/Thelia/Tests/Action/ProductTest.php

```diff
@@ -396,4 +413,533 @@ public function testDelete(ProductModel $product)
 
         $this->assertTrue($deletedProduct->isDeleted());
     }
+
+    /*
+     * Cloning process tests
+     */
+
+    public function testCreateClone()
+    {
+        $kernel = new Thelia('test', true);
+        $kernel->boot();
+
+        $originalProduct = ProductQuery::create()->addAscendingOrderByColumn('RAND()')->findOne();
+        $newRef = uniqid('testClone-');
+
+        $event = new ProductCloneEvent($newRef, 'fr_FR', $originalProduct);
+        $event->setDispatcher($kernel->getContainer()->get('event_dispatcher'));
+
+        $originalProductDefaultI18n = ProductI18nQuery::create()->findPk([$originalProduct->getId(), $event->getLang()]);
+        $originalProductDefaultPrice = ProductPriceQuery::create()->findOneByProductSaleElementsId($originalProduct->getDefaultSaleElements()->getId());
+
+        // Call function to test
+        $action = new Product();
+        $action->createClone($event, $originalProductDefaultI18n, $originalProductDefaultPrice);
+
+        $cloneProduct = $event->getClonedProduct();
+
+        // Check product information
+        $this->assertInstanceOf(
+            'Thelia\Model\Product',
+            $cloneProduct,
+            'Instance of clone product must be Thelia\Model\Product'
+        );
+
+        $this->assertFalse($cloneProduct->isNew(), 'IsNew must be false');
+        $this->assertEquals('fr_FR', $cloneProduct->getLocale(), 'Locale must be equal');
+        $this->assertEquals($originalProductDefaultI18n->getTitle(), $cloneProduct->getTitle(), 'Title must be equal');
+        $this->assertEquals($newRef, $cloneProduct->getRef(), 'Ref must be equal');
+        $this->assertEquals(0, $cloneProduct->getVisible(), 'Visible must be false');
+        $this->assertEquals($originalProduct->getDefaultCategoryId(), $cloneProduct->getDefaultCategoryId(), 'Default categories must be equal');
+
+        $this->assertGreaterThan(0, $cloneProduct->getPosition(), 'Position must be greater than 0');
+
+        $clonedProductSaleElements = $cloneProduct->getProductSaleElementss();
+
+        $this->assertEquals(1, count($clonedProductSaleElements), 'There is not only one default PSE (maybe more, maybe none)');
```

Comment:
You can use ```$this->assertCount``` https://phpunit.de/manual/current/en/appendixes.assertions.html#appendixes.assertions.assertCount